src/AppChooser/Dialog.vala

```diff
@@ -122,51 +121,38 @@ public class AppChooser.Dialog : Hdy.Window {
 
         carousel = new Hdy.Carousel () {
             allow_long_swipes = true,
-            allow_mouse_drag = true,
-            expand = true
+            allow_mouse_drag = true
         };
 
         var switcher = new Hdy.CarouselIndicatorDots () {
             carousel = carousel
         };
 
-        var cancel = new Gtk.Button.with_label ("Cancel");
-        var select = new Gtk.Button.with_label ("Select");
-        select.get_style_context ().add_class (Gtk.STYLE_CLASS_SUGGESTED_ACTION);
-
-        var button_box = new Gtk.ButtonBox (Gtk.Orientation.HORIZONTAL) {
-            layout_style = Gtk.ButtonBoxStyle.END,
-            valign = Gtk.Align.END,
-            margin_top = 12,
-            expand = true,
-            spacing = 6
+        var cancel = new Gtk.Button.with_label ("Cancel") {
+            halign = Gtk.Align.END
         };
 
-        button_box.add (cancel);
-        button_box.add (select);
-
         var grid = new Gtk.Grid () {
             orientation = Gtk.Orientation.VERTICAL,
             column_spacing = 12,
             row_spacing = 6,
             margin = 12
         };
-
         grid.attach (overlay, 0, 0, 1, 2);
         grid.attach (primary_label, 1, 0);
         grid.attach (secondary_label, 1, 1);
-        grid.attach (carousel, 0, 3, 2);
-        grid.attach (switcher, 0, 4, 2);
-        grid.attach (button_box, 1, 5);
+        grid.attach (carousel, 0, 2, 2);
+        grid.attach (switcher, 0, 3, 2);
+        grid.attach (cancel, 1, 4);
```

Comment:
without the ButtonBox, the button is really small. i wonder if we can put the switcher in the ButtonBox and have both on the same row.